### DISCUSSION TITLE
Moving the agent with its state

### DISCUSSION THREAD

**Discussion by eldadts** [aldinokemal/go-whatsapp-web-multidevice#141] [General]
Hi,
I need to move the agent from EC2 to EKS (k8s) but I want to keep its state (no QR code rescan)
Is it enough to copy `whatapp.db` to the new instance or are there any other files that should be also copied?

**Comment by aldinokemal** [comment]
Short answer: copy the whole `storages/` directory, not just the single `.db` file, and copy it with the process stopped.

- `storages/whatsapp.db` is the whatsmeow device/session store (`src/config/settings.go:41`, `src/infrastructure/whatsapp/database.go:16-42`). That is the file that saves you the QR rescan.
- It is opened in WAL mode (`src/infrastructure/whatsapp/database.go:35` -> `src/pkg/sqlite/sqlite_cgo.go:14-29`), so `whatsapp.db-wal` and `whatsapp.db-shm` can still hold writes that are not in the main file yet. Stop the container first, then copy, or copy all three files together.
- If you set `DB_KEYS_URI`, that database holds the session/encryption key cache and must be copied as well (`src/cmd/root.go:671-673`, `readme.md:236`).
- Since v7.10.0 there is also `storages/chatstorage.db` for chat and message history (`src/config/settings.go:76`). Not required for login, only if you want history and reply context to survive the move.

The container mounts the whole directory at `/app/storages` (`docker-compose.yml:18`), so on EKS: one PVC mounted at `/app/storages`, pre-seeded with the contents of the EC2 `storages/` dir.

Two things to watch on k8s:

1. Run a single replica (StatefulSet, ReadWriteOnce PVC). The store is a local SQLite file and two pods sharing one WhatsApp session will fight over it.
2. If you prefer Postgres instead of a PVC, `DB_URI=postgres://user:pass@host:5432/db` is supported (`src/infrastructure/whatsapp/database.go:37-38`), but there is no sqlite to postgres migration in the project, so switching storage backends does mean a fresh QR scan.
